x-pack/lib/config_management/elasticsearch_source.rb

```diff
@@ -35,8 +35,16 @@ class RemoteConfigError < LogStash::Error; end
 
       def initialize(settings)
         super(settings)
-        if @settings.get("xpack.management.enabled") && !@settings.get_setting("xpack.management.elasticsearch.password").set?
-          raise ArgumentError.new("You must set the password using the \"xpack.management.elasticsearch.password\" in logstash.yml")
+        if @settings.get("xpack.management.enabled")
+          if @settings.get_setting("xpack.management.elasticsearch.cloud_id").set?
+            if !@settings.get_setting("xpack.management.elasticsearch.cloud_auth").set?
+              raise ArgumentError.new("You must set credentials using \"xpack.management.elasticsearch.cloud_auth\" in logstash.yml")
```

Comment:
Consider adding "when using `xpack.management.elasticsearch.cloud_id`" to the error message